src/App.tsx

```diff
@@ -1,17 +1,100 @@
-import React from 'react';
-import './App.css';
+import { FC, useState } from 'react';
+import './App.scss';
 
+import todos from './api/todos';
 import users from './api/users';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+const preparedTodos: Todo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
+
+const App: FC = () => {
+  const [inputedTitle, setInputedTitle] = useState('');
+  const [choosedUser, setCoosedUser] = useState('');
+  const [isTryAddTodo, setIsTryAddTodo] = useState(false);
+
+  const addNewTodo = () => {
+    const lastTodo = preparedTodos[preparedTodos.length - 1];
+    const findedUser = users.find(user => user.name === choosedUser);
+
+    if (inputedTitle.length > 0 && choosedUser !== 'Choose a user') {
+      preparedTodos.push({
+        userId: findedUser?.id || 0,
+        id: lastTodo.id + 1,
+        title: inputedTitle,
+        user: findedUser || null,
+        completed: false,
+      });
```

Comment:
no, don't mutate array, save this to your state and use a construction like this `[...todos, newTodo]`